Fix `EditorContextMenuPlugin` `CONTEXT_SLOT_FILESYSTEM_CREATE` does not receive path information for some menus

OzelotVanilla opened the pull request:
Testable based on the code of [this pull request](https://github.com/godotengine/godot/pull/106811), since this bug was found when fixing another issue.

fix #106819

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

KoBeWi commented on editor/filesystem_dock.cpp:
```diff
@@ -2605,7 +2605,14 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
+					if (!fpath.ends_with("/")) {
+						fpath = fpath.get_base_dir();
+					}
+					Vector<String> paths;
+					paths.push_back(fpath);
+					EditorContextMenuPluginManager::get_singleton()->activate_custom_option(EditorContextMenuPlugin::CONTEXT_SLOT_FILESYSTEM_CREATE, p_option, paths);
```
```suggestion
					EditorContextMenuPluginManager::get_singleton()->activate_custom_option(EditorContextMenuPlugin::CONTEXT_SLOT_FILESYSTEM_CREATE, p_option, { fpath });
```

OzelotVanilla commented:
Sorry I just mis-typed the commit message !

OzelotVanilla commented:
Typo fixed.

OzelotVanilla commented:
> I realized this also requires updating the docs.

I found the editing the document will make conflict when this PR and [another one](https://github.com/godotengine/godot/pull/106811) merge to the master branch.
Should I solve merge conflicts by myself, or just finish updating the document here ?

KoBeWi commented:
Update the doc normally and then you'll have to rebase the other PR after one is merged.

OzelotVanilla commented:
> Update the doc normally and then you'll have to rebase the other PR after one is merged.

Got it. Waiting for either PR to be merged first.

Repiteo commented:
Could you squash the commits? See our [workflow guidelines](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#modifying-a-pull-request) for more information

OzelotVanilla commented:
> Could you squash the commits? See our [workflow guidelines](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#modifying-a-pull-request) for more information

OK. I misunderstood the content. Let me squash all into one.

OzelotVanilla commented:
Now my two PRs has done the squashing job. Thank you all for the patience and instructions.
Waiting for either PR to be merged first.

Repiteo commented:
That PR got merged first, so this'll need that rebase

OzelotVanilla commented:
Just finished the rebase work and pushed.
And I remember in the first, I wrongly chose `4.4` branch for testing, then I changed to `master`, found that the path param for the callback function changes.

In my own opinion, I also agree that changing the behaviour of passing the param for the callback is better.

Repiteo commented:
Is that compilation error tied to this PR, or is it unrelated?

OzelotVanilla commented:
> Is that compilation error tied to this PR, or is it unrelated?

I would like to say this PR does not breaks the compilation.
Previously this PR is based on branch `4.4`, and I had successfullt compiled this many times.

I notice that `drivers/metal` is only generating warnings while compiling `4.4` but errors for `master`. I guess that the changing in that folder makes the compilation fail.

Warning example while compiling `4.4`:
```
drivers/metal/pixel_formats.mm:576:41: warning: mixture of designated and non-designated initializers in the same initializer list is a C99 extension [-Wc99-designator]
  576 |         _mtl_pixel_format_descs[p_pix_fmt] = { .mtlPixelFormat = p_pix_fmt, DataFormat::DATA_FORMAT_MAX, p_fmt_caps, p_view_class, p_pix_fmt_linear, p_name };
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/metal/pixel_formats.mm:576:70: note: first non-designated initializer is here
  576 |         _mtl_pixel_format_descs[p_pix_fmt] = { .mtlPixelFormat = p_pix_fmt, DataFormat::DATA_FORMAT_MAX, p_fmt_caps, p_view_class, p_pix_fmt_linear, p_name };
  ```

OzelotVanilla commented:
By the way, I think this pull request itself could be merged for now @Repiteo.

OzelotVanilla committed:
fix: EditorContextMenuPlugin CONTEXT_SLOT_FILESYSTEM_CREATE does not receive path information for some menu

Also add explanation in `EditorContextMenuPlugin`'s doc for fixing path information passing

(Accept PR #106820 suggestion by: Tomasz Chabora <kobewi4e@gmail.com>)

OzelotVanilla commented:
> Thanks!

Thank you also !
